CSharp/DevVmPowershell/Helpers/SqlHelper.cs

```diff
@@ -122,5 +123,94 @@ INNER JOIN
 				throw new Exception("Error Enabling Data Grid for the Extracted Text Field", ex);
 			}
 		}
+
+		public bool CreateOrAlterShrinkDbProc()
+		{
+			bool wasSuccessful = false;
+			Console.WriteLine($"{nameof(CreateOrAlterShrinkDbProc)} - Creating or Altering {Constants.SqlScripts.ShrinkDbProcName}");
+
+			string sql = Constants.SqlScripts.CreateOrAlterShrinkDbProc;
+
+			if (DoesSqlObjectExist(Constants.SqlScripts.SchemaName, Constants.SqlScripts.ShrinkDbProcName))
+			{
+				Console.WriteLine($"{nameof(CreateOrAlterShrinkDbProc)} - {Constants.SqlScripts.ShrinkDbProcName} exists so changing statement to ALTER");
+				sql = sql.Replace("CREATE PROCEDURE", "ALTER PROCEDURE");
+			}
+
+			try
+			{
+				DbContext.ExecuteNonQuerySQLStatement(sql);
+				wasSuccessful = true;
+				Console.WriteLine($"{nameof(CreateOrAlterShrinkDbProc)} - Successfully updated {Constants.SqlScripts.ShrinkDbProcName}");
+			}
+			catch (Exception ex)
+			{
+				Console.WriteLine($"{nameof(CreateOrAlterShrinkDbProc)} - Failed to create or alter {Constants.SqlScripts.ShrinkDbProcName}");
+				Console.WriteLine(ex);
+			}
+
+			return wasSuccessful;
+		}
+
+		public bool RunShrinkDbProc()
+		{
+			bool wasSuccessful = false;
+			Console.WriteLine($"{nameof(RunShrinkDbProc)} - About to Run {Constants.SqlScripts.ShrinkDbProcName}");
+
+			try
+			{
+				if (DoesSqlObjectExist(Constants.SqlScripts.SchemaName, Constants.SqlScripts.ShrinkDbProcName))
+				{
+					using (SqlConnection connection = connectionHelper.GetSqlConnection(Constants.Connection.Sql.ConnectionString_ConnectTimeoutLong))
+					{
+						Console.WriteLine($"{nameof(RunShrinkDbProc)} - Running {Constants.SqlScripts.ShrinkDbProcName}...");
+						SqlCommand command = new SqlCommand(Constants.SqlScripts.ExecuteShringDbProc, connection);
+						command.CommandTimeout = connection.ConnectionTimeout;
+						command.Connection.Open();
+						command.ExecuteNonQuery();
+					}
+
+					wasSuccessful = true;
+					Console.WriteLine($"{nameof(RunShrinkDbProc)} - Successfully ran {Constants.SqlScripts.ShrinkDbProcName}");
+				}
+				else
+				{
+					Console.WriteLine($"{nameof(RunShrinkDbProc)} - Failed to run {Constants.SqlScripts.ShrinkDbProcName} as it doesn't exist");
+				}
+			}
+			catch (Exception ex)
+			{
+				Console.WriteLine($"{nameof(RunShrinkDbProc)} - Failed to run {Constants.SqlScripts.ShrinkDbProcName}");
+				Console.WriteLine(ex);
+			}
+
+			return wasSuccessful;
+		}
+
+		private bool DoesSqlObjectExist(string schema, string objectName)
+		{
+			bool exists = false;
+			string sql = Constants.SqlScripts.DoesExist.Replace("@@schema", schema).Replace("@@objectName", objectName);
+
+			Console.WriteLine($"{nameof(DoesSqlObjectExist)} - Checking if object exists {Constants.SqlScripts.ShrinkDbProcName}");
+
+			try
+			{
+				object queryResult = DbContext.ExecuteSqlStatementAsScalar(sql);
```

Comment:
you could call ExecuteSqlStatementAsScalar<int>(sql) method instead of calling the non-generic method and then converting the return value to int later.

Reply:
changed to ExecuteSqlStatementAsScalar<int>